### ISSUE TITLE
Hotkeys stopped working since Spotify v1.0.75.483.g7ff4a0dc

### ISSUE THREAD

**Issue by aleab** [aleab/toastify#31]
The underlying issue is that Spotify changed the hierarchy of its windows and their names.

![Spy++](https://vgy.me/LeNQUZ.png)

![Spy++](https://vgy.me/cNlMru.png)

Before version 1.0.75.483.g7ff4a0dc, the hierarchy was:
- "Spotify" SpotifyMainWindow
  * CefBrowserWindow
    + Chrome_WidgetWin_0

[These](https://github.com/aleab/toastify/blob/5f75d8135973f5c3728c4eb04cec1fb587ac3cc1/Toastify/src/Core/Spotify.cs#L420-L437) are the lines of code that need to be fixed, i.e. the methods `GetMainWindowHandle` and `GetCefWidgetWindowHandle` in the `Spotify` class.

**Comment by aleab** [comment]
## MEMO

***Complex* hotkeys that required the use of Window Messages (see [`SendShortcut`](https://github.com/aleab/toastify/blob/fa4f35624ee62216813cf88229b4077075248e65/Toastify/src/Core/Spotify.cs#L585-L654)) stopped working!**

The issue is that the new main window (*"Spotify" Chrome_WidgetWin_0*) is not a regular window with a **static** menu bar anymore, so the *"Playback"* submenu's items cannot be reached in a simple, programmatical way anymore (at least not with `GetMenu` > `GetSubMenu`).

The following window messages (irrelevant ones are filtered out) can be captured when clicking the new smartphone-app-like three-dots button in the top-left corner (which is just another *"window"*, or more like a CEF widget component, with class name *Chrome_WidgetWin_0* and empty title).

![Spy++](https://vgy.me/1tJDiC.png)

From what I gathered, a brand new window (surely the dropdown menu) is created each time the top-left button is clicked (obviously, each time with a different handle).
As expected, both calling the [`GetMenu`](https://msdn.microsoft.com/en-us/library/windows/desktop/ms647640.aspx) function from the Windows APIs and sending a [MN_GETHMENU](https://msdn.microsoft.com/en-us/library/windows/desktop/ms632613.aspx) message resulted in a null pointer, since there's no static menu bar.

**Comment by aleab** [comment]
Fixed in the [latest release](https://github.com/aleab/toastify/releases/tag/v1.10.3).

**Comment by abelgomez** [comment]
One question: I assume that with the latest changes in 
9cb83d1 (dropping support for "complex" actions), implementing hotkeys for switching the Shuffle/Repeat toggles is no longer possible, isn't it?

**Comment by aleab** [comment]
@abelgomez, mmmhhh... it surely is not possible anymore by using the *"hacks"* I used for VolumeUp/VolumeDown and FastForward/Rewind (although, I did found an alternative method for FastForward/Rewind in the end).

**However**, it is possible to implement them in some other way, either by using Spotify's WebAPIs (#17) or by using another *hacky* method, which consists in "navigating" Spotify's page (which is an HTML web page) using some WebDriver and "clicking" on the Shuffle/Repeat toggles.
I honestly dislike both methods. The former would require users to authenticate to Spotify with their credentials from time to time (since WebAPIs require authentication); the latter would be performance-heavy.

We'll see.
